### PR TITLE
Add node location to SQL tree nodes

### PR DESCRIPTION
**Pull request by takezoe**

### CODE REVIEW

File: airframe-sql/src/main/scala/wvlet/airframe/sql/model/ResolvedPlan.scala [hunk 1]
```diff
@@ -24,11 +24,20 @@ import wvlet.airframe.sql.model.LogicalPlan.Relation
...
     columns.map { col =>
-      ResolvedAttribute(col.name, col.dataType, None, Some(table), Some(col))
+      ResolvedAttribute(
+        col.name,
+        col.dataType,
+        None,
+        Some(table),
+        Some(col),
+        None
+      ) // TODO This doesn't allow to have NodeLocation
```
Comment (takezoe) [inline airframe-sql/src/main/scala/wvlet/airframe/sql/model/ResolvedPlan.scala:40]: `ResolvedAttribute` always has no `NodeLocation`.

File: airframe-sql/src/main/scala/wvlet/airframe/sql/model/ResolvedPlan.scala [hunk 2]
```diff
@@ -24,11 +24,20 @@ import wvlet.airframe.sql.model.LogicalPlan.Relation
...
     columns.map { col =>
-      ResolvedAttribute(col.name, col.dataType, None, Some(table), Some(col))
+      ResolvedAttribute(
+        col.name,
+        col.dataType,
+        None,
+        Some(table),
+        Some(col),
+        None // ResolvedAttribute always has no NodeLocation
+      )
```
Comment (takezoe) [inline airframe-sql/src/main/scala/wvlet/airframe/sql/model/ResolvedPlan.scala:40]: `ResolvedAttribute` always has no `NodeLocation`

Comment (xerial) [reply airframe-sql/src/main/scala/wvlet/airframe/sql/model/ResolvedPlan.scala:40]: It's ok for now. TypeResolve needs to propagate NodeLocation wherever possible

File: airframe-sql/src/main/scala/wvlet/airframe/sql/parser/SQLInterpreter.scala [hunk 3]
```diff
@@ -29,7 +29,7 @@ object SQLInterpreter {
 /**
   * ANTLR parse tree -> SQL LogicalPlan
   */
-class SQLInterpreter extends SqlBaseBaseVisitor[Any] with LogSupport {
+class SQLInterpreter(withNodeLocation: Boolean = true) extends SqlBaseBaseVisitor[Any] with LogSupport {
```
Comment (takezoe) [inline airframe-sql/src/main/scala/wvlet/airframe/sql/parser/SQLInterpreter.scala:32]: Making it possible to build AST without NodeLocation mainly for testing.

File: airframe-sql/src/main/scala/wvlet/airframe/sql/model/LogicalPlan.scala [hunk 4]
```diff
@@ -386,7 +397,12 @@ object LogicalPlan {
       columnNames match {
         case Some(aliases) =>
           query.outputAttributes.zip(aliases).map { case (in, alias) =>
-            SingleColumn(in, Some(alias))
+            SingleColumn(
+              in,
+              Some(alias),
+              None,
+              alias.nodeLocation // TODO Is alias.nodeLocation suitable as NodeLocation for this?
```
Comment (takezoe) [inline airframe-sql/src/main/scala/wvlet/airframe/sql/model/LogicalPlan.scala:404]: 🤔

Comment (xerial) [reply airframe-sql/src/main/scala/wvlet/airframe/sql/model/LogicalPlan.scala:404]: It's not accurate, but returning the close position would be good enough.

**Comment by xerial** [comment]
LGTM. Added a location parameter to the SQLError class for properly reporting the error location.

**Pull request merged**